libmozevent/mercurial.py

```diff
@@ -396,6 +396,21 @@ def get_status():
             )
             await asyncio.sleep(TRY_STATUS_DELAY)
 
+    def is_eligible_for_retry(self, error):
+        """
+        Given a Mercurial error message, if it's an error likely due to a
+        temporary connection problem, consider it as eligible for retry.
+        """
+        eligible_errors = [
+            "push failed on remote",
+            "stream ended unexpectedly",
+            "error: EOF occurred in violation of protocol",
+        ]
+        for eligible_message in eligible_errors:
+            if eligible_message in error:
+                return True
+        return False
```

Comment:
nit: since you are refactoring this part, using `any()` would be more pythonic 
```suggestion
        return any(eligible_message in error for eligible_message in eligible_errors)
```